[DOCS] Data quality on integrity

Quantisan opened the pull request:
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

Quantisan committed:
DQ integrity.md draft

intro, sample data, expectations, example

flesh out sample data and Expectations

update intro

update Data preview

update Key Expectations section

flesh out cross-table example

reduce Scenarios section

update Scenarios

conclusion

copy edit Key Expectations intro

rachhouse commented on docs/docusaurus/docs/reference/learn/data_quality_use_cases/integrity.md:
````diff
@@ -0,0 +1,278 @@
+2. `transfer_balances` table ([CSV](https://raw.githubusercontent.com/great-expectations/great_expectations/develop/tests/test_sets/learn_data_quality_use_cases/integrity_transfer_balances.csv)):
+
+```
+| transaction_id | transfer_amount | adjustment | sender_debit | recipient_credit |
````
I think it would be helpful to distinguish between the two datasets by having different pk columns - is there any reason we wouldn't want this column to be `transfer_id`?

Quantisan replied:
e7430282754462fb5f4c4e5b60e9ef4dc0cab149

rachhouse commented on docs/docusaurus/docs/reference/learn/data_quality_use_cases/integrity.md:
````diff
@@ -0,0 +1,278 @@
+1. `transactions` table ([CSV](https://raw.githubusercontent.com/great-expectations/great_expectations/develop/tests/test_sets/learn_data_quality_use_cases/integrity_transactions.csv)):
+
+```
+| transaction_id | sender_id | recipient_id | amount  | transaction_date | reference_number | confirmation_code |
````
What's the different between `reference_number` and `confirmation_code`?

Quantisan replied:
while these values have different semantic meanings, they happen to be numerically equivalent in this example. i'm using this overlap to demonstrate integrity constraints in a simplified single-table scenario.

rachhouse commented on docs/docusaurus/docs/reference/learn/data_quality_use_cases/integrity.md:
```diff
@@ -0,0 +1,278 @@
+
+Verifies that values in one column are consistently greater than related values in another column.
+
+**Use Case**: Ensure transfer amounts are always greater than or equal to adjustments.
```
My knowledge of banking is only based on my experience as a consumer - would an adjustment ever be larger than a transfer amount in the real world? For my own edification for working on this article, what are common reasons to have an adjustment on a transfer?

Quantisan replied:
One example that I can think of: if the transfer amount is tiny and there's some fee or whatever, then yes, adjustment could be greater than the transfer amount.

Some reasons to have an adjustment on a transfer:
1. backdated transactions
2. fee
3. exchange rate adjustment
4. customer service; e.g. goodwill credit, dispute resolution

rachhouse commented on docs/docusaurus/docs/reference/learn/data_quality_use_cases/integrity.md:
```diff
@@ -0,0 +1,278 @@
+   defaultValue="gx_cloud"
+   values={[
+      {value: 'gx_cloud', label: 'GX Cloud'},
+      {value: 'gx_core', label: 'GX Core'}
```
I haven't had a chance to dig into the GX Core side of this, but I'll take the action on writing the code example.

Quantisan committed:
add a transfer_id column to the transfer_balances table

https://github.com/great-expectations/great_expectations/pull/10583#discussion_r1821602187

rachhouse committed:
Merge branch 'develop' into docs/data-quality-integrity-2

rachhouse committed:
add integrity to landing page and TOC

rachhouse committed:
remove future .gif path so netlify preview builds for now

rachhouse committed:
stub out tests

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Quantisan committed:
moved Expectations to CI

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rachhouse committed:
update integrity data and develop custom SQL Expectation example

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rachhouse committed:
typo and path fixes

rachhouse committed:
Merge branch 'develop' into docs/data-quality-integrity-2

rachhouse committed:
add integrity expectations in public_api_missing_threshold check

rachhouse committed:
what do you want docusaurus

rachhouse committed:
Merge branch 'develop' into docs/data-quality-integrity-2

rachhouse committed:
Merge branch 'develop' into docs/data-quality-integrity-2

rachhouse committed:
rework examples to explicitly show use of built-in expectation vs custom SQL expectation

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

rachhouse committed:
Merge branch 'develop' into docs/data-quality-integrity-2

rachhouse reviewed (approved):
I've incorporated feedback from internal GX review (conducted via slack) and am approving the PR so it can get merged in.

rachhouse committed:
Merge branch 'develop' into docs/data-quality-integrity-2
